packages/cli/src/commands/lint.ts

```diff
@@ -3,36 +3,54 @@ import {
   formatProblems,
   getTotals,
   lint,
-  loadConfig,
+  lintConfig,
+  getConfigFromString,
+  findConfig,
   getMergedConfig,
   OutputFormat,
+  makeDocumentFromString,
 } from '@redocly/openapi-core';
 import {
   getExecutionTime,
   getFallbackEntryPointsOrExit,
   handleError,
   pluralize,
   printLintTotals,
+  printConfigLintTotals,
   printUnusedWarnings,
+  readFileAsString,
 } from '../utils';
 import { Totals } from '../types';
 import { blue, gray, red } from 'colorette';
 import { performance } from 'perf_hooks';
 
-export async function handleLint(
-  argv: {
-    entrypoints: string[];
-    'max-problems'?: number;
-    'generate-ignore-file'?: boolean;
-    'skip-rule'?: string[];
-    'skip-preprocessor'?: string[];
-    extends?: string[];
-    config?: string;
-    format: OutputFormat;
-  },
-  version: string,
-) {
-  const config: Config = await loadConfig(argv.config, argv.extends);
+type LintOptions = {
+  entrypoints: string[];
+  'max-problems'?: number;
+  'generate-ignore-file'?: boolean;
+  'skip-rule'?: string[];
+  'skip-preprocessor'?: string[];
+  'skip-config-lint'?: boolean;
+  extends?: string[];
+  config?: string;
+  format: OutputFormat;
+};
+
+export async function handleLint(argv: LintOptions, version: string) {
+  const configPath = findConfig(argv.config);
+  const configContent = configPath ? await readFileAsString(configPath) : '';
+
+  if (configPath && shouldValidateLintConfig(argv)) {
+    await validateLintConfig({
+      configContent,
+      configPath,
+      outputFormat: argv.format,
+      maxProblems: argv['max-problems'],
+      version,
+    });
+  }
+
+  const config: Config = await getConfigFromString(configContent, configPath, argv.extends);
```

Comment:
To me it looks not good, because `loadConfig` should do the only thing it is supposed to do. And most probably there are libraries that import `loadConfig` which expect to get config and not to have it linted (and probably thrown lint results into console)

Reply:
We may want to use config linting in other places where we load config. To prevent unwanted logs, we may pass an optional third parameter into `loadConfig` to indicate whether to lint config while loading.